pkg/kubefed2/join.go

```diff
@@ -357,18 +356,27 @@ func registerCluster(crClientset *crclient.Clientset, clusterNamespace, host, jo
 		return cluster, nil
 	}
 
-	cluster, err := crClientset.ClusterregistryV1alpha1().Clusters(clusterNamespace).Create(cluster)
-	if idempotent && errors.IsAlreadyExists(err) {
-		glog.V(2).Infof("Cluster %s already exists in the cluster registry", joiningClusterName)
-		return cluster, nil
+	existingCluster, err := crClientset.ClusterregistryV1alpha1().Clusters(clusterNamespace).Get(cluster.Name, metav1.GetOptions{})
+	switch {
+	case err != nil && !errors.IsNotFound(err):
+		glog.V(2).Infof("Cannot retrieve cluster registry cluster %s due to %v", cluster.Name, err)
+		return nil, err
+	case err == nil && errorOnExisting:
+		return nil, fmt.Errorf("cluster registry cluster %s already exists", cluster.Name)
```

Comment:
We've recently switched to `github.com/pkg/errors`, and there is a ci check that will fail if `fmt.Errorf` is used:

https://github.com/kubernetes-sigs/federation-v2/pull/563

(No action required) A refactor of join/unjoin should likely replace the info logging with error wrapping.

cc: @gyliu513 (since we've discussed this before)